### PR TITLE
Remove string "Array" funnybusiness

### PR DESCRIPTION
**Pull request by kshyatt**
I think this looks much nicer and the tests seem to pass. ~~I didn't get all of them but I did get most.~~ Now got 'em all!

### CODE REVIEW

File: test/linalg/lu.jl [hunk 1]
```diff
@@ -69,23 +69,16 @@ dimg  = randn(n)/2
...
-                    @test norm(a'*(lua'\b) - b, 1) < ε*κ*n*2 # Two because the right hand side has two columns
+            let Bs = copy(b), Cs = copy(c)
+                for (bb, cc) in ((Bs, Cs), (view(Bs, 1:n, 1), view(Cs, 1:n)))
+                    @test norm(a*(lua\bb) - bb, 1) < ε*κ*n*2 # Two because the right hand side has two columns
+                    @test norm(a'*(lua'\bb) - bb, 1) < ε*κ*n*2 # Two because the right hand side has two columns
                     @test norm(a'*(lua'\a') - a', 1) < ε*κ*n^2
-                    @test norm(a*(lua\c) - c, 1) < ε*κ*n # c is a vector
-                    @test norm(a'*(lua'\c) - c, 1) < ε*κ*n # c is a vector
+                    @test norm(a*(lua\cc) - cc, 1) < ε*κ*n # c is a vector
+                    @test norm(a'*(lua'\cc) - cc, 1) < ε*κ*n # c is a vector
```
Comment (fredrikekre) [inline test/linalg/lu.jl]: `# cc is a vector` and above.

**Review by fredrikekre — APPROVED** [review]

File: test/linalg/lu.jl (hunk 1 above)
Comment (kshyatt) [reply test/linalg/lu.jl]: oh good call, thanks <3

**Review by StefanKarpinski — APPROVED** [review]
Wow, that was some serious fishiness.

**Comment by kshyatt** [comment]
@andreasnoack thoughts?

File: test/linalg/dense.jl [hunk 2]
```diff
@@ -169,13 +147,7 @@ end # for eltya
 
 @testset "test triu/tril bounds checking" begin
     ainit = rand(5,7)
-    for arraytype in ("Array", "SubArray")
-        if arraytype == "Array"
-            a = ainit
-        else
-            a = view(ainit, 1:size(ainit, 1), 1:size(ainit, 2))
-        end
-        @test_throws(ArgumentError,triu(a,8))
```
Comment (fredrikekre) [inline test/linalg/dense.jl]: Accidentally removed this test?

Comment (kshyatt) [reply test/linalg/dense.jl]: The Return Of The Test!

**Comment by fredrikekre** [comment]
Timeout on bsd and libgit2 on circle. Linalg tests passed on all workers.

**Comment by andreasnoack** [comment]
LGTM

**Pull request merged**